provisioning/miniprov/hare_mp/main.py

```diff
@@ -254,21 +255,202 @@ def test(args):
+    if int(resp[2]):
+        logging.error('hctl failed to updated the status %s', resp[1])
+        raise Exception(
+            f'Command {hctl_status} exited with error code {int(resp[2])}.'
+            f'Command output: {resp[1]}')
+    else:
+        if 'Cluster is not running' not in resp[0]:
+            resp = executecmds(hctl_shutdown)
+            logging.info('hctl shutdown: %s', resp[0])
+
+        logging.info('-------Starting BOOTSTRAP-SHUTDOWN in LOOP-------')
+        for count in range(loop_count):
+            logging.info('Loop count# {}'.format(count + 1))
+            now = datetime.now()     # current date and time
+            date_time = now.strftime("%Y-%m-%d %H:%M:%S")
+            jlog = '~/journal_ctrl_' + now.strftime('%Y_%m_%d_%H%M%S') + '.log'
+
+            logging.info('Start hctl Bootstrap')
+            resp = bootstrap_cluster(str(args.file[0]), True)
+            if resp:
+                logging.error('Failed to bootstrap')
+                resp = executecmds(['journalctl', '--since', date_time, '>',
+                                    jlog])
+                logging.info('created journal log %s', jlog)
+                exit(-1)
+
+            logging.info('Check that all the services are up in hctl.')
+            if is_cluster_running():
+                logging.info('hctl is running.')
+            else:
+                logging.error('Still hctl is not running.')
+                resp = executecmds(['journalctl', '--since', date_time, '>',
+                                    jlog])
+                logging.info('created journal log %s', jlog)
+                exit(-1)
+
+            sleep(10)
+            logging.info('Shutdown the cluster.')
+            resp = executecmds(hctl_shutdown)
+            logging.info('hctl shutdown: %s', resp[0])
+            if int(resp[2]):
+                logging.error('Shutdown Failed %s', resp[1])
+                resp = executecmds(
+                    ['journalctl', '--since', date_time, '>', jlog])
+                logging.info('created journal log.%s', jlog)
+                raise Exception(
+                    f'Command {hctl_shutdown} exited with error code '
+                    f'{int(resp[2])}. Command output: {resp[1]}')
+
+            sleep(10)
+            if is_cluster_running():
+                logging.error('Still Cluster is running.')
+                resp = executecmds(['journalctl', '--since', date_time, '>',
+                                    jlog])
+                logging.info('created journal log %s', jlog)
+                exit(-1)
+
+    test_hare_postreq(str(args.file[0]), date_time, jlog)
+
+
 def test_IVT(args):
     try:
         rc = 0
         path_to_cdf = args.file[0]
+        is_dev_opt_enbl = int(args.dev[0])
 
-        logging.info('Running test plan: ' + str(args.plan[0].value))
+        if not is_dev_opt_enbl:
+            logging.info('Running test plan: ' + str(args.plan[0].value))
```

Comment:
Log message is repeated both in `if` and in `else` branches. Let's move it out of `if` statment.